WordPress/Classes/ViewRelated/Jetpack/Branding/Banner/UIScrollView+JPScrollViewDataDelegate.swift

```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+extension UIScrollView: JPScrollViewDataDelegate {
+
+    var translationY: CGFloat {
+        return panGestureRecognizer.translation(in: superview).y
+    }
+
+    var scrollViewFrameHeight: CGFloat {
+        frame.height
+    }
+
+    var scrollViewContentHeight: CGFloat {
+        contentSize.height
+    }
+
+    var scrollViewContentOffsetY: CGFloat {
+        contentOffset.y + adjustedContentInset.top
+    }
```

Comment:
Feels a bit odd having `UIScrollView` confirming to `UIScrollViewDelegate`, what do you think?

Reply:
sure, but if you look at its definition:
`protocol JPScrollViewDelegate: UIScrollViewDelegate`
you can see how it inherits from `UIScrollViewDelegate`

Reply:
Oh nvm, you changed the definition of the protocol as well. In any case, I still think that:
- The view (`JerpackBannerView`) should not be responsible of making all those assumptions about when the container wants to show it or not
- We should not send a reference to a `UIScrollView` instance as an event.
- One way to avoid repeating code (how many viewControllers will need this specialized behavior)? could be to inject the algorithm into the view itself in form of a closure that returns a boolean, and `JetpackBannerView` can use that boolean for the hide/show animation. If this proves too challenging, we can even add an extension for now that contains all the calculations, but we should still avoid passing a reference to the scroll view. We can pass a dictionary of values, if we really need this calculation not to happen at the sender site.

Reply:
Also, since now it's not a scrollViewDelegate anymore, I believe it should be named differently